src/App.tsx

```diff
@@ -14,19 +40,31 @@ const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setSort={setSort}
+                setQuery={setQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {onLoading && (
+                <Loader />
+              )}
+              <TodoList
+                todos={filteredTodos}
+                setSelectedTodo={setSelectedTodo}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      {false && (
-        <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          setSelectedTodo={setSelectedTodo}
```

Comment:
This callback for closing TodoModal
```suggestion
          onClose={setSelectedTodo}
```